src/TRestRawSignalRecoverChannelsProcess.cxx

```diff
@@ -175,6 +175,8 @@ TRestEvent* TRestRawSignalRecoverChannelsProcess::ProcessEvent(TRestEvent* evInp
         // cout << "Channel id : " << fChannelIds[x] << " Left : " << idL << " Right
         // : " << idR << endl;
 
+        fOutputSignalEvent->RemoveSignalWithId(fChannelIds[x]);
```

Comment:
why to remove this signal from the event? In principle I would expect that the signal id you want to recover is not present inside the event. So I would check first if the signal is found inside the event:

```
if( fOutputSignalEvent->GetSignalIndex(fChannelIds[x])> 0 ) fOutputSignalEvent->RemoveSignalWithId(fChannelIds[x]);
```

Reply:
Right, I should put the check first.

In case we read out all channels in the daq, we also get noise signals in the disconnected (dead) channels. In this case REST gives an warning `Warning. Signal ID : 4541 already exists. Signal will not be added to signal event`.